### DISCUSSION TITLE
How to communicate with Redux/MobX/Other State management tools elegantly.

### DISCUSSION THREAD

**Discussion by wewoor** [DTStack/molecule#676] [Ideas]

**Comment by hepezu** [comment]
In my case, I create a service to collect the events and states, and notify the redux. I like the current state management, although it is a little bit strange to some react developers. It is flexible enough, more detailed documentation on it would be nice.

**Comment by wewoor** [comment]
> In my case, I create a service to collect the events and states, and notify the redux. I like the current state management, although it is a little bit strange to some react developers. It is flexible enough, more detailed documentation on it would be nice.

The events and states are in one service, is the service too complex? We separate the States and Events via the Controller layer.

**Comment by hepezu** [comment]
I mainly use your mode to manage states(service and controller with global DI) in project developed with molecule, and I like that. 

I have some modules with old codes which use redux, so I make a service to do the communication between existing redux module and services in molecule(not all service states, only what I need). I love the current molecule's architecture, by using extensions and molecule services, it is facile to make extension modules with different tech stacks.

**Comment by wewoor** [comment]
> I mainly use your mode to manage states(service and controller with global DI) in project developed with molecule, and I like that.
> 
> I have some modules with old codes which use redux, so I make a service to do the communication between existing redux module and services in molecule(not all service states, only what I need). I love the current molecule's architecture, by using extensions and molecule services, it is facile to make extension modules with different tech stacks.

Your feedback is very important to us. we are planning to extract this architecture into a separate package, and it can be used for other React projects. Maybe we can keep in touch, lol~ 

https://github.com/DTStack/molecule/discussions/622

**Comment by hepezu** [comment]
Happy Spring Festival.

Sure, keep in touch, I would like to make a demo in the future.
